qa/rpc-tests/feelimit.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Dogecoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Fee limit QA test.
+# Tests nodes under fee limit, verifies fee rounding
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+from decimal import Decimal
+
+class FeeLimitTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+
+    def setup_network(self, split=False):
+        self.nodes = []
+        self.nodes.append(start_node(0, self.options.tmpdir, []))
+        self.nodes.append(start_node(1, self.options.tmpdir, []))
+
+        connect_nodes_bi(self.nodes,0,1)
+
+        self.is_network_split=False
+        self.sync_all()
+
+    def run_test(self):
+        # mine some blocks
+        self.nodes[0].generate(101)
+        self.sync_all()
+
+        # Output address
+        addr_to = self.nodes[1].getnewaddress()
+        lower_fee_per_kb = Decimal("0.001")
+
+        # Force generating a transaction with 1.14.5-like fees by manually building the tx
+        utx = self.nodes[0].listunspent()[0]
+        inputs  = [ {'txid' : utx['txid'], 'vout' : utx['vout']}]
+        outputs = { addr_to : utx['amount'] - Decimal("1.0") }
+        rawtx   = self.nodes[0].createrawtransaction(inputs, outputs)
+
+        # Work out how big this transaction would be, then change the output to match
+        tx_size = count_bytes(self.nodes[0].signrawtransaction(rawtx)['hex'])
+        fee = lower_fee_per_kb * tx_size / Decimal("1000")
```

Comment:
This still gives me a larger utxo size after signing the second time a little over 1 out of 4 times, would recommend to just add a byte for variance so this can pass, saves a loop.

Reply:
https://gist.github.com/patricklodder/2b778c4d6449d3b70100efcc41667865 if you want to check it for yourself.

Reply:
I've added a "data" output so I can control the tx size fairly precisely, and that's committed now. Let me know what you think?

Reply:
That doesn't help with the signature length variance 😕

In the wallet code we loop until we get it right, which is the "proper" fix, it just triggers less often when you always round up, and the quickfiix alternative is to always add a byte to the size calc (because the variant is really only one byte, so at most that would be `2/1000*fee_rate` deviation from recommended fee.